edb/pgsql/delta.py

```diff
@@ -4358,46 +4351,97 @@ def _compile_conversion_expr(
             src_path_id = ptr_path_id.src_path()
             assert src_path_id
 
-            if ptr_table and not orig_rel_is_always_source:
-                rvar = compiler.new_external_rvar(
-                    rel_name=(alias,),
-                    path_id=ptr_path_id,
-                    outputs={
-                        (src_path_id, ('identity',)): 'source',
-                    },
-                )
-                external_rvars[ptr_path_id, 'source'] = rvar
-                external_rvars[ptr_path_id, 'value'] = rvar
-                external_rvars[src_path_id, 'identity'] = rvar
-                if local_table_only and not is_lprop:
-                    external_rvars[src_path_id, 'source'] = rvar
-                    external_rvars[src_path_id, 'value'] = rvar
-                elif is_lprop:
-                    external_rvars[tgt_path_id, 'identity'] = rvar
-                    external_rvars[tgt_path_id, 'value'] = rvar
-            else:
-                src_rvar = compiler.new_external_rvar(
-                    rel_name=(alias,),
+            external_rels = {
+                src_path_id: compiler.new_external_rel(
+                    rel_name=(source_alias,),
                     path_id=src_path_id,
-                    outputs={},
                 )
-                external_rvars[src_path_id, 'identity'] = src_rvar
-                external_rvars[src_path_id, 'value'] = src_rvar
-                external_rvars[src_path_id, 'source'] = src_rvar
-        else:
-            external_rvars = None
+            }
+
+            assert isinstance(ir.expr, irast.Set)
+
+            # Wrap the expression into a select with iterator, so DML and
+            # volatile expressions are executed once for each object.
+
+            # IR ast wrapping
+            typ = orig_schema.get(f'schema::ObjectType', type=s_types.Type)
+            outer_path = irast.PathId.from_type(
+                orig_schema, typ, typename=sn.QualName("std", "obj")
+            )
+
+            scope_iter = irast.ScopeTreeNode(
+                unique_id=10001,
```

Comment:
Let's use negative numbers. (And bind them to names so we don't mention each number more than once.)